example/example_test.go

```diff
@@ -70,6 +71,56 @@ Content-Type: application/json
 				{Url: "query string protobuf", SomeSnakedName: 2},
 			},
 		},
+		ExpectedHeadersDump: `HTTP/1.1 200 OK
+Content-Length: 31
+Content-Type: application/x.prottp;<*example.SearchResponse>
+
+`,
+	},
+
+	{
+		Name:   "test request json, response x.prottp normal",
+		URL:    "/example.SearchService/Search",
+		Accept: "application/x.prottp, application/json;q=0.9, */*;q=0.8",
+		JSONReqBody: `{
+			"query": "query string protobuf",
+			"page_number": 1,
+			"result_per_page": 10
+		}`,
+		ExpectedPBResBody: &example.SearchResponse{
+			Result: []*example.Result{
+				{Url: "query string protobuf", SomeSnakedName: 2},
+			},
+		},
+		ExpectedHeadersDump: `HTTP/1.1 200 OK
+Content-Length: 31
+Content-Type: application/x.prottp;<*example.SearchResponse>
```

Comment:
Sorry, another unclear description on my part.

With 

```
Content-Type: application/json;<package name.message name>
```

`<>` were intended as placeholders, not as part of the content.

Also, I was wrong about the "parameter" (bit after `;`), it needs to be `<key>=<value>` (and key cannot start with `q`...).

So instead of this:

```
Content-Type: application/x.prottp;<*example.SearchResponse>
```

I was asking for it to be this:

```
Content-Type: application/x.prottp;name=example.SearchResponse
```

I'm flexible with `name`. Could be `messageName`, `ms`, `type`, `message`, etc...

And it looks like you can get the message name via [`proto.MessageName`](https://godoc.org/github.com/golang/protobuf/proto#MessageName) so that would be better than using the Go type name. Especially since messages are pointers!

Reply:
Updated.